tasks/apm-inject-install.yml

```diff
@@ -0,0 +1,39 @@
+---
+- name: Include APM host injection Debian install tasks
+  include_tasks: pkg-debian/install-apm-inject-latest.yml
+  when: not ansible_check_mode and ansible_facts.os_family == "Debian" and datadog_apm_instrumentation_enabled != ""
+
+- name: Include APM host injection RedHat install tasks
+  include_tasks: pkg-redhat/install-apm-inject-latest.yml
+  when: not ansible_check_mode and ansible_facts.os_family in ["RedHat", "Rocky", "AlmaLinux"] and datadog_apm_instrumentation_enabled != ""
+
+#- name: Check if dd-host-install needs to run
+#  command: dd-host-install --dry-run
+#  register: datadog_apm_instrumentation_rc
+#  ignore_errors: true
+
+- name: Run APM host injection setup script
+  command: dd-host-install --no-config-change --no-agent-restart
+  notify: restart datadog-agent
+  when: not ansible_check_mode and datadog_apm_instrumentation_rc == 1 and datadog_apm_instrumentation_enabled == "host"
+
+#- name: Check if dd-host-container-install needs to run
+#  command: dd-host-install --dry-run && dd-container-install --dry-run
+#  register: datadog_apm_instrumentation_rc
+#  ignore_errors: true
+
+- name: Run APM host-docker injection (Docker) setup script
+  # dd-host-container-install will also reload docker as it changes /etc/docker/daemon.json
+  command: dd-host-install --no-agent-config-change --no-agent-restart && dd-container-install --no-config-change --no-agent-restart
+  notify: restart datadog-agent
+  when: not ansible_check_mode and datadog_apm_instrumentation_rc == 1 and datadog_apm_instrumentation_enabled == "all"
```

Comment:
Instead of having this task, why don't we run the other two with e.g. `when: datadog_apm_instrumentation_enabled in ["all", "host"]` and `when: datadog_apm_instrumentation_enabled in ["all", "docker"]`?

Reply:
sounds good! changed 👍